### PR TITLE
Improvements to Linux packaging

### PR DESCRIPTION
**Pull request by guihkx**
These changes are meant to make things easier for package maintainers to distribute Notes on their Linux distribution.

We now use CMake to configure *and* install:

- The [desktop file](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html)
- The [AppStream metadata file](https://freedesktop.org/software/appstream/docs/chap-Metadata.html) (added just now!)
- ... And of course, our app icons in [many resolutions](https://github.com/nuttyartist/notes/tree/fe80934e8625bfbd72c5892305e2b44b2f9cd1ae/packaging/linux/common/icons)!

Now, Linux package maintainers won't have to look for those files manually, or perhaps tweak them before packaging Notes: It should be all done automagically!

The build steps for packaging should be as easy as:

```shell
$ cmake -B builddir -DCMAKE_INSTALL_PREFIX=/usr \
    -DCMAKE_BUILD_TYPE=Release \
    -DUPDATE_CHECKER=OFF 
$ cmake --build builddir
$ DESTDIR="package_dir" cmake --install builddir
```

P.S.: There are some changes unrelated to Linux here as well, but since they're so minor, I decided to include them here instead of opening a new PR. Hope it's okay.... :)

### CODE REVIEW

File: packaging/linux/common/notes.appdata.xml.in
Comment (guihkx) [inline packaging/linux/common/notes.appdata.xml.in:10]: @nuttyartist Is this your actual, current email? I actually don't remember where I got it from... 😓

File: packaging/linux/common/notes.appdata.xml.in
Comment (guihkx) [inline packaging/linux/common/notes.appdata.xml.in:31]: Note to self: Update these screenshots once @nuttyartist take new ones.

File: packaging/linux/common/notes.appdata.xml.in
Comment (guihkx) [inline packaging/linux/common/notes.appdata.xml.in:56]: Note to self: I'm assuming `2.1.0` gets released today, but if it doesn't, make sure to update the `date` property here.

File: packaging/linux/common/notes.appdata.xml.in
Comment (guihkx) [inline packaging/linux/common/notes.appdata.xml.in:61]: I don't *love* having to use `<code>` to represent keyboard keys here, but unfortunately, using `<kbd>` makes `appstreamcli` fail to validate this file. 🤷

File: packaging/linux/common/notes.appdata.xml.in
Comment (nuttyartist) [reply packaging/linux/common/notes.appdata.xml.in:10]: Yes! It is!

File: packaging/linux/common/notes.appdata.xml.in
Comment (nuttyartist) [reply packaging/linux/common/notes.appdata.xml.in:31]: 👍

File: packaging/linux/common/notes.appdata.xml.in
Comment (nuttyartist) [reply packaging/linux/common/notes.appdata.xml.in:56]: If everything goes well, then tomorrow (:

**Comment by guihkx** [comment]
If someone wants to inspect the contents of the AppImage we're building, you can extract it with:

```
$ ./Notes*.AppImage --appimage-extract
```

And then you can browse the `squashfs-root/usr` directory to look at the files we're shipping. :)

**Comment by nuttyartist** [comment]
Great.

> The build steps for packaging should be as easy as:

Should we add this to README?

BTW, I want to disclose md5 hashes of the binaries created by AppVeyor that we're going to put in under the Release. Maybe we can automatically generate them? I guess for the macOS version, it currently doesn't make sense since I will need to sign and notarize it on my own machine.

Another question, on release (pushing to the main) the binaries' names not gonna have the commit id, etc attached to them, right? 

Another question, Is the flatpack also going to be auto-generated via Actions as well?

**Comment by guihkx** [comment]
>Should we add this to README?

I'm actually working on documenting the build steps (and build options) across Windows, macOS and Linux, but it's quite extensive and a bit messy, so it's not ready yet...

>BTW, I want to disclose md5 hashes of the binaries created by Github Actions that we're going to put in under the Release. Maybe we can automatically generate them? I guess for the macOS version, it currently doesn't make sense since I will need to sign and notarize it on my own machine.

This is possible, but in my opinion it should be done once we get the notarization and signing done completely in CI too.

Once that is achieved, we can even use a GitHub "release action" to automate the releasing process even further, and then the checksums can be generated there as well.

>Another question, on release (pushing to the main) the binaries' names not gonna have the commit id, etc attached to them, right?

Once you push a tag (e.g. `v2.1.0`), then GitHub Actions will actually trigger a new build. And this build will not have the commit id attached to the app version.

>Another question, Is the flatpack also going to be auto-generated via Actions as well?

No, the Flatpak manifest will be maintained on separate repository, under the `flathub` namespace on GitHub, where we will have commit access.

For example, the Flatpak manifest for LibreOffice is maintained here:

https://github.com/flathub/org.libreoffice.LibreOffice

And regarding its maintainability, it should be very easy.

Don't worry, you'll get to see it in action once the app is submitted to Flathub. :)

**Comment by nuttyartist** [comment]
>I'm actually working on documenting the build steps (and build options) across Windows, macOS and Linux, but it's quite extensive and a bit messy, so it's not ready yet...

Okay, we'll wait.

> This is possible, but in my opinion it should be done once we get the notarization and signing done completely in CI too.

> Once that is achieved, we can even use a GitHub "release action" to automate the releasing process even further, and then the checksums can be generated there as well.

Alrighty, sounds like a cool plan (:

> Once you push a tag (e.g. v2.1.0), then GitHub Actions will actually trigger a new build. And this build will not have the commit id attached to the app version.

👍 

> No, the Flatpak manifest will be maintained on separate repository, under the flathub namespace on GitHub, where we will have commit access.

Oh, ok. Nice.

**Comment by guihkx** [comment]
This didn't get any formal reviews, but I really want to include these changes before v2.1.0 is released.

@nuttyartist Do I have your approval?

**Comment by nuttyartist** [comment]
It's good to have someone else also review this. I think we can push this before 2.1.0 is released.

**Review by zjeffer — APPROVED** [review]

**Comment by nuttyartist** [comment]
Still trying to figure out some issues with notarizing Notes (Getting invalid status over `Unnotarized Developer ID`). So delaying it for at least tomorrow.

**Comment by nuttyartist** [comment]
I believe we can merge this. And we can release 2.1.0 today.

**Comment by bjorn** [comment]
I won't have time to give this a detailed review and it's a bit outside of my area anyway since I'm not really familiar with CMake, so please don't wait on me. :-)

**Review by nuttyartist — APPROVED** [review]

**Pull request merged**
